src/ring_theory/nilpotent.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import data.nat.choose.sum
+
+/-!
+# Nilpotent elements
+
+## Main definitions
+
+  * `is_nilpotent`
+  * `is_nilpotent_neg_iff`
+  * `commute.is_nilpotent_add`
+  * `commute.is_nilpotent_mul_left`
+  * `commute.is_nilpotent_mul_right`
+  * `commute.is_nilpotent_sub`
+
+## Tags
+
+nilpotent, subhas_mul, subhas_add
+-/
+
+universes u
+
+variables {R : Type u} {x y : R}
+
+/-- An element is said to be nilpotent if some natural-number-power of it equals zero.
+
+Note that we require only the bare minimum assumptions for the definition to make sense. Even
+`monoid_with_zero` is too strong since nilpotency is important in the study of rings that are only
+power-associative. -/
+def is_nilpotent [has_zero R] [has_pow R ℕ] (x : R) : Prop := ∃ (n : ℕ), x^n = 0
+
+lemma monoid_with_zero.pow_zero_of_le [monoid_with_zero R] {n m : ℕ} (hn : n ≤ m) (hx : x^n = 0) :
+  x^m = 0 :=
+by rw [← nat.sub_add_cancel hn, pow_add, hx, mul_zero]
+
+lemma is_nilpotent_neg_iff [ring R] : is_nilpotent (-x) ↔ is_nilpotent x :=
+begin
+  suffices : ∀ (y : R), is_nilpotent y → is_nilpotent (-y),
+  { refine ⟨_, this x⟩, intros h, rw ← neg_neg x, exact this (-x) h, },
+  rintros y ⟨n, hn⟩,
+  use n,
+  rw [neg_pow, hn, mul_zero],
+end
```

Comment:
I think this `suffices` is worth having as a separate `is_nilpotent.neg` definition,